### ISSUE TITLE
Improper decoding of MNZ instruction

### ISSUE THREAD

**Issue by Boolsheet** [Textmode/TMVM#13]
It seems to me that his function has some errors.

<code>machine.lua:158
    [0x0e]=function(self) --  MNZ RET, R, &nn # free-register conditional move to literal address
        local a, b = convreg(self, self.memory[self.IP+1])

```
    if a then
        self.memory[self.memory[self.IP+1]] = self[b]
    end
    return 3;
end;
```

</code>

'if a then' will always be true.<br>
And on the next line, shouldn't that be 'self.IP+2'?

**Comment by Textmode** [comment]
Oh...oh wow...thats very, embarrassingly wrong!

Fixed, but not closing until I've had a chance to confirm the fix (created issue #14)

**Comment by Textmode** [comment]
The opcode passes the test-case.
